src/lib/support/JniReferences.cpp

```diff
@@ -388,4 +388,23 @@ CHIP_ERROR JniReferences::GetObjectField(jobject objectToRead, const char * name
     return err;
 }
 
+jstring JniReferences::CharToJniString(const char * value)
+{
+    JNIEnv * env        = GetEnvForCurrentThread();
+    jobject jbyteBuffer = env->NewDirectByteBuffer((void *) value, static_cast<jlong>(strlen(value)));
+
+    jclass charSetClass = env->FindClass("java/nio/charset/Charset");
+    jmethodID charsetForNameMethod =
+        env->GetStaticMethodID(charSetClass, "forName", "(Ljava/lang/String;)Ljava/nio/charset/Charset;");
+    jobject charsetObject = env->CallStaticObjectMethod(charSetClass, charsetForNameMethod, env->NewStringUTF("UTF-8"));
+
+    jmethodID charSetDecodeMethod = env->GetMethodID(charSetClass, "decode", "(Ljava/nio/ByteBuffer;)Ljava/nio/CharBuffer;");
```

Comment:
`decode` says `This method always replaces malformed-input and unmappable-character sequences with this charset's default replacement byte array` ... is this really acceptable instead of failing decode?

Reply:
This code is simply designed to prevent crashes.
It seems that the current discussion is discussing that it is correct to return an error in the case of an invalid string byte.

In this case, it is necessary to add a code that checks whether it can be changed to utf-8 before decoding and transmits it as an error.

Reply:
It has been modified to return as an error when it is not decoded as `UTF-8`.
(In a file created with zap, it will be returned as null.)

Reply:
In addition, in the decode method of the CharsetDecoder class, an exception occurs when decoding cannot be performed.
Refer https://docs.oracle.com/javase/7/docs/api/java/nio/charset/CharsetDecoder.html#decode(java.nio.ByteBuffer)